openedx/core/djangoapps/content_staging/api.py

```diff
@@ -1,4 +1,100 @@
 """
 Public python API for content staging
 """
-# Currently, there is no public API.
+from __future__ import annotations
+from datetime import datetime
+from typing import NamedTuple
+
+from django.http import HttpRequest
+from opaque_keys.edx.keys import UsageKey
+
+from .models import UserClipboard as _UserClipboard, StagedContent as _StagedContent
+from .serializers import UserClipboardSerializer as _UserClipboardSerializer
+
+
+StagedContentStatus = _StagedContent.Status
+CLIPBOARD_PURPOSE = _UserClipboard.PURPOSE
+
+
+class StagedContentData(NamedTuple):
+    """ Read-only data model for StagedContent """
+    id: int
+    user_id: int
+    created: datetime
+    purpose: str
+    status: StagedContentStatus
+    block_type: str
+    display_name: str
+
+
+class UserClipboardData(NamedTuple):
+    """ Read-only data model for StagedContent """
+    content: StagedContentData
+    source_usage_key: UsageKey
```

Comment:
Per [OEP-49](https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0049-django-app-patterns.html#data-py), please make these `attr` classes in `data.py` unless you have a strong reason for preferring named tuples (and put comments explaining that decision).

Reply:
I like NamedTuples since they're built-in and have all the functionality we need in this case, but it's no problem at all - I'll change this to use attrs and live in data.py for consistency.

Reply:
I've moved them to data.py though I found some issues with the OEP-49 recommendation regarding enums - https://github.com/openedx/open-edx-proposals/pull/471 - so I'm sticking with Django enums for now.